lib/winpki/ldap.go

```diff
@@ -49,19 +78,29 @@ type LDAPConfig struct {
 	ServerName string
 	// CA is an optional CA cert to be used for verification if InsecureSkipVerify is set to false.
 	CA *x509.Certificate
+	// LocateServer contains parameters for locating LDAP servers from the AD domain.
+	LocateServer
+	// Logger is the logger for the service.
+	Logger *slog.Logger
 }
 
-// Check verifies this LDAPConfig
-func (cfg LDAPConfig) Check() error {
-	if cfg.Addr == "" {
-		return trace.BadParameter("missing Addr in LDAPConfig")
+// CheckAndSetDefaults verifies this LDAPConfig
+func (cfg *LDAPConfig) CheckAndSetDefaults() error {
+	cfg.Logger = cmp.Or(cfg.Logger, slog.With(teleport.ComponentKey, teleport.ComponentWindowsDesktop))
```

Comment:
Technically this code is shared, not specific to windows desktops. We should not default  to `ComponentWindowsDesktop`. Maybe use `"ldap"` there instead.

Also, a practice I like to follow for these `CheckAndSetDefaults` methods is to do the error checks first, and only start setting things after the errors are covered. This way we're not mutating `cfg` if there was an error.